src/http/routes.cc

```diff
@@ -53,7 +53,7 @@ routes::~routes() {
             delete r.second;
         }
     }
-
+    delete _default_handler;
```

Comment:
but we don't claim that the ownership of the handler is transformed from the caller of `add_default_handler()` to the `routes` instance. instead of deleting this pointer, i'd suggest revise the doxygen comment of `add_default_handler()` to explain that the callee does not take the ownership of this pointer, the caller is responsible to keep it alive.

this change would break the existing seastar applications. for instance, see https://github.com/redpanda-data/redpanda/blob/ecf2e99c97f5fb60d8971ce4e4760724c2b3cf02/src/v/cloud_storage/tests/s3_imposter.cc#L509

Reply:
reverted